src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "firmware/shared/math/polynomial_2d.h"
+#include "firmware/shared/math/vector_2d.h"
+
+#define __TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS__                                          \
+    6000  // The maximum size of the array containing trajectory elements
+#define __TRAJECTORY_PLANNER_MAX_PATH_LENGTH__ 18  // [meters]
+
+// Struct that defines a single point on a trajectory
+// Includes the Position,and Timedata corresponding to that point
+typedef struct TrajectoryElement
+{
+    Vector2d_t position;
+    double time;
+} TrajectoryElement_t;
+
+typedef struct Trajectory
+{
+    TrajectoryElement_t* trajectory_elements;
+    unsigned int num_elements;
+} Trajectory_t;
```

Comment:
That sounds like a good idea. The issue I see here is if the path is small the rounding error introduced by a massive number of elements might be an issue.

Also in the case of the constant-time trajectory, where Trajectory_t is used we cannot just fill up the array because the path will finish before that.

Reply:
I'll change it so that the trajectory array is an input parameter.

Reply:
Done

Reply:
Looks like we still are doing static memory allocation on line 191 of `trajectory_planner.c`? So we'll still have the issues I mentioned above.

Reply:
Removed static array allocation from function